src/elements/content-sidebar/Sidebar.js

```diff
@@ -91,14 +96,68 @@ class Sidebar extends React.Component<Props, State> {
 
         this.state = {
             isDirty: this.getLocationState('open') || false,
+            isDocGenTemplate: false,
         };
 
         this.setForcedByLocation();
     }
 
+    disableDocgenTab() {
+        const { history, location } = this.props;
+        this.setState({
+            ...this.state,
+            isDocGenTemplate: false,
+        });
+        if (location.pathname === `/${SIDEBAR_VIEW_DOCGEN}`) {
+            history.push('/');
+        }
+    }
+
+    fetchMetadataSuccessCallback = ({ editors }: { editors: Array<MetadataEditor> }) => {
+        const { history } = this.props;
+        if (editors && editors.length > 0) {
+            const template = editors[0]?.template;
+            const scopeIsGlobal = template.scope === 'global';
+            const templateKeyIsBoxDocGen = template.templateKey === 'boxDocGen';
+            const isDocGenTemplate = scopeIsGlobal && templateKeyIsBoxDocGen;
+
+            if (isDocGenTemplate) {
+                this.setState({ ...this.state, isDocGenTemplate });
+                // navigate to docgen tab by default
+                history.push(`/${SIDEBAR_VIEW_DOCGEN}`);
+            }
+        } else {
+            this.disableDocgenTab();
+        }
+    };
+
+    fetchMetadata(): void {
```

Comment:
Ok so my biggest concern here is the overall approach. We're treating the docgen side bar panel different from the rest of them e.g. acitivty, versions, metadata etc as those are controlled by props passed into the component and not by the component iteself making an API call to determine whether or not the docgen panel should be shown or not. Is there a rreason why we decided to go this route instead of having the consumer of the sidebar pass something along the lines of hasDocGenTemplate to the ContentSidebar that is set based on the API calls you've added here?

Reply:
indeed we have tickets in place for getting this information from preview calls in monolith instead of making a separate call to metadata from FE. I used this approach here in order to confirm the functionality working and unblock the e2e testing work. I can move the logic into preview page FE but eventually we'll need to remove it for performance reasons there as well to avoid unnecessary calls to metadata api from frontend if we can leverage some call preview team is making (that is if it is making a call to metadata api) on file view.

But you're right it'd be better to move the call logic into consumer instead, will do

Reply:
I removed the metadata fetching logic out of BUIE - now it's done in the client. Later, we'll remove the call to 'checkDocGenTemplate' completely